modules/3_classification/src/classifier.cpp

```diff
@@ -11,11 +11,38 @@ using namespace cv::utils::fs;
 void topK(const std::vector<float>& src, unsigned k,
           std::vector<float>& dst,
           std::vector<unsigned>& indices) {
-    CV_Error(Error::StsNotImplemented, "topK");
+
+    std::vector<std::pair<int, float*>> srcIndexed;
+
+    for (int index = 0; index < src.size(); index++) {
+        srcIndexed.push_back(std::pair<int, float*>(index, &const_cast<float &>(src[index])));
+    }
+
+    std::sort(
+            srcIndexed.begin(),
+            srcIndexed.end(),
+            [](std::pair<int, float*> i, std::pair<int, float*> j) -> bool { return *i.second > *j.second; }
+            );
+
+    k = srcIndexed.size() < k ? srcIndexed.size() : k;
+
+    for (int index = 0; index < k; index++) {
+        indices.push_back(srcIndexed[index].first);
+        dst.push_back(*srcIndexed[index].second);
+    }
 }
 
 void softmax(std::vector<float>& values) {
-    CV_Error(Error::StsNotImplemented, "softmax");
+    float sumOfExps = 0;
+    float maxValue = *std::max_element(values.begin(), values.end());
+
+    for (auto value: values) {
+        sumOfExps += exp(value / maxValue);
```

Comment:
The formula is incorrect - you cannot divide by max value

Reply:
Why not? This is done to lower the sumOfExps (for the test with high values).
Besides, both softmax tests are succeed.

Reply:
Original formula is

<img src="https://render.githubusercontent.com/render/math?math=y_{i} = \frac{e^{x_i}}{\sum_{j=1}^{N} e^{xj}}">

but your is

<img src="https://render.githubusercontent.com/render/math?math=(\frac{e^{x_i}}{\sum_{j=1}^{N} e^{xj}})^{\frac{1}{max(x_{i})}}">

Reply:
Thanks for provement, fixed